portal/src/hook/usePivot.ts

```diff
@@ -0,0 +1,24 @@
+import { useCallback } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { IPivotItemProps } from "@fluentui/react";
+
+export function usePivot(): {
+  hash: string;
+  onLinkClick: (item?: { props: IPivotItemProps }) => void;
+} {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const hash = location.hash.slice(1);
+
+  const onLinkClick = useCallback(
+    (item) => {
+      const itemKey = item?.props.itemKey;
+      if (typeof itemKey === "string") {
+        navigate(`./#${itemKey}`);
```

Comment:
Please escape the fragment for safety.

Reply:
> I tried checking/unchecking the `Activate` column in `Authenticator` tab, and it seems it doesn't count as modified. Is that intentional?

This is not expected, after investigation this is cause by mutating the state in setState directly also mutate initial state as they shared same reference